pkg/apis/portworx/v1beta2/types.go

```diff
@@ -0,0 +1,75 @@
+package v1beta2
+
+import (
+	"github.com/portworx/talisman/pkg/apis/portworx/v1beta1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumePlacementStrategy specifies a spec for volume placement in the cluster
+type VolumePlacementStrategy struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            VolumePlacementSpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumePlacementStrategyList is a list of VolumePlacementStrategy objects
+type VolumePlacementStrategyList struct {
+	meta.TypeMeta `json:",inline"`
+	meta.ListMeta `json:"metadata,omitempty"`
+	// Items are the list of volume placements strategy items
+	Items []VolumePlacementStrategy `json:"items"`
+}
+
+// VolumePlacementSpec specifies a set of rules for volume placement in the cluster
+type VolumePlacementSpec struct {
+	// Rules defines a list of rules as part of the placement spec. All the rules specified will
+	// be applied for volume placement.
+	// Rules that have enforcement as "required" are strictly enforced while "preferred" are best effort.
+	// In situations, where 2 or more rules conflict, the weight of the rules will dictate which wins.
+	Rules VolumePlacementRule `json:"rules"`
```

Comment:
This should have been an array, right?
Else call it just `rule` or probably remove this indirection and pull all the *Affinity in the VolumePlacementSpec directly.

Reply:
good idea. Simplified this and removed the unnecessary rules layer.